.github/workflows/black.yml

```diff
@@ -0,0 +1,33 @@
+name: python auto-format
+on:
+  # Triggers the workflow on push to all the branches and on pull request to the "master" or the "dev" branch
+  push:
+  pull_request:
+    branches: [ "master", "dev"]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install black
```

Comment:
I read that a best practice of installing multiple packages is to do them all in one line:

py -m pip install black black[Jupyter]

That way it only needs to resolve the conflicts/dependencies once, it's doesn't matter much in this case as it's just 2 packages, but just a hint to keep in the back of the head.